src/frontier/frontier_cmd_arcade.c

```diff
@@ -0,0 +1,526 @@
+    }
+    
+    BattleSetup_Delete(setup);
+    
+    return FALSE;
+}
+
+BOOL FrtCmd_ArcadeStartBattle(FrontierContext *ctx) {
+    FrontierLaunchParam *param = Frontier_GetLaunchParam(ctx->frontierSystem->unk0);
+    ArcadeContext *arcadeData = Frontier_GetData(ctx->frontierSystem->unk0);
+    BattleSetup *setup = BattleArcade_NewBattleSetup(arcadeData, param);
+
+    arcadeData->battleSetup = setup;
+    
+    Frontier_LaunchApplication(ctx->frontierSystem->unk0, &gOverlayTemplate_Battle, setup, 0, NULL);
+
+    return TRUE;
+}
+
+static void GameBoardArgs_Set(GAME_BOARD_ARGS *args, ArcadeContext *data) {
+    args->type = data->type;
+    args->unk1E = ov80_02238498(data);
+    
+    args->unk14 = &data->unk20;
+    args->unk10 = &data->unk13;
+    args->bpGain = data->bpGain;
+    
+    args->winstreak = data->winstreak;
+    args->multiWinstreak = data->multiWinstreak;
+
+    args->cursorSpeed = &data->cursorSpeed;
+    args->playerParty = data->playerParty;
+    args->opponentParty = data->opponentParty;
+    args->work = data;
+
+    data->weather = 0;
+    args->weather = &data->weather;
+    args->randomFlag = data->randomFlag;
+    data->randomFlag = 0;
+    data->unk1F = 0;
+
+    int partyCnt = Party_GetCount(data->playerParty);
+
+    for (int i = 0; i < partyCnt; i++) {
+        Pokemon *mon = Party_GetMonByIndex(data->playerParty, i);
+
+        data->savedHp[i] = GetMonData(mon, MON_DATA_MAXHP, NULL);
+        data->savedAtk[i] = GetMonData(mon, MON_DATA_ATK, NULL);
+        data->savedDef[i] = GetMonData(mon, MON_DATA_DEF, NULL);
+        data->savedSpd[i] = GetMonData(mon, MON_DATA_SPEED, NULL);
+        data->savedSpAtk[i] = GetMonData(mon, MON_DATA_SPATK, NULL);
+        data->savedSpDef[i] = GetMonData(mon, MON_DATA_SPDEF, NULL);
+    }
+}
+
+static void ov80_02233A1C(void *data) {
+    GAME_BOARD_ARGS *args = data;
+    ov80_02234550(args->work, args);
+    FreeToHeap(args);
+}
+
+BOOL FrtCmd_ArcadeSetPartyBeforeBattle(FrontierContext *ctx) {
+    ArcadeContext *arcadeCtx = Frontier_GetData(ctx->frontierSystem->unk0);
+    BattleArcade_SetPartyBeforeBattle(arcadeCtx);
+    return FALSE;
+}
+
+BOOL FrtCmd_ArcadeSetPartyAfterBattle(FrontierContext *ctx) {
+    ArcadeContext *arcadeCtx = Frontier_GetData(ctx->frontierSystem->unk0);
+    BattleArcade_SetPartyAfterBattle(arcadeCtx);
+    return FALSE;
+}
+
+BOOL FrtCmd_ArcadeAction(FrontierContext *ctx) {
+    Pokemon *mon;
+    Party *party;
+    int i;
+    PlayerProfile *profile; 
+    
+    u32 action = FrontierScript_ReadShort(ctx);
+    u8 var0 = FrontierScript_ReadVar(ctx);
+    u8 var1 = FrontierScript_ReadVar(ctx);
+    u16 *out = FrontierScript_ReadVarPtr(ctx);
+    ArcadeContext *arcadeCtx = Frontier_GetData(ctx->frontierSystem->unk0);
+    FrontierLaunchParam *param = Frontier_GetLaunchParam(ctx->frontierSystem->unk0);
+    FrontierMap *frontierMap = FrontierSystem_GetFrontierMap(ctx->frontierSystem);
+
+    switch (action) {
+        case 2:
```

Comment:
action consts?

Reply:
`case` labels should be flush with the `switch`

Reply:
I'll do action constants when I get to dumping the scripts cause I'm pretty sure they're shared for all the frontier gimmicks